sstable/src/dictionary.rs

```diff
@@ -524,6 +637,222 @@ mod tests {
         (dictionary, table)
     }
 
+    #[test]
+    fn test_term_to_ord_or_next() {
+        let dict = {
+            let mut builder = Dictionary::<MonotonicU64SSTable>::builder(Vec::new()).unwrap();
+
+            builder.insert(b"bbb", &1).unwrap();
+            builder.insert(b"ddd", &2).unwrap();
+
+            let table = builder.finish().unwrap();
+            let table = Arc::new(PermissionedHandle::new(table));
+            let slice = common::file_slice::FileSlice::new(table.clone());
+
+            Dictionary::<MonotonicU64SSTable>::open(slice).unwrap()
+        };
+
+        assert_eq!(dict.term_ord_or_next(b"aaa").unwrap(), TermOrdHit::Next(0));
+        assert_eq!(dict.term_ord_or_next(b"bbb").unwrap(), TermOrdHit::Exact(0));
+        assert_eq!(dict.term_ord_or_next(b"bb").unwrap(), TermOrdHit::Next(0));
+        assert_eq!(dict.term_ord_or_next(b"bbbb").unwrap(), TermOrdHit::Next(1));
+        assert_eq!(dict.term_ord_or_next(b"dd").unwrap(), TermOrdHit::Next(1));
+        assert_eq!(dict.term_ord_or_next(b"ddd").unwrap(), TermOrdHit::Exact(1));
+        assert_eq!(dict.term_ord_or_next(b"dddd").unwrap(), TermOrdHit::Next(2));
+
+        // Shouldn't this be u64::MAX?
+        assert_eq!(
+            dict.term_ord_or_next(b"zzzzzzz").unwrap(),
+            TermOrdHit::Next(2)
+        );
```

Comment:
for very small sstables (only one block), we don't store an index, and the pseudo-index always reply that the answer lies in block number 0. After reading the whole block, `decode_up_to_or_next` conclude that the key is after its own end

Reply:
Thanks for the clarification

Reply:
can you update the comment?

Reply:
yep, done